CRD cannot be generated for Unstructured type

We got a type that has `*unstructured.Unstructured` as [one of its fields](https://github.com/crossplaneio/crossplane/blob/master/pkg/apis/workload/v1alpha1/types.go#L150) and we're generating CRD off of it. I am working on transition to controller-tools v0.2.0 and running into the following error:
```
github.com/crossplaneio/crossplane/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go:45:2: encountered struct field "Object" without JSON tag in type "Unstructured"
```

Looked into the source code and there really isn't a tag for `Object` field, however, even if I manually write the json tag, it returns the following:
```
github.com/crossplaneio/crossplane/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go:45:20: map values must be a named type, not *ast.InterfaceType
```

Is there a way to have controller-tools handle this? I imagine it's more complicated due to having `interface{}` as value but I'd appreciate if we can find a way for it to work. Note that it worked with version 0.1.9

You can see the ongoing work here https://github.com/crossplaneio/crossplane/pull/665

You can see the versions used in `Gopkg.toml`.

To reproduce, checkout the PR and run `make manifests`

You can also try to have `*unstructured.Unstructured` typed field in any struct and try to generate CRD.

--- a/pkg/crd/known_types.go
+++ b/pkg/crd/known_types.go
@@ -68,6 +68,13 @@ var KnownPackages = map[string]PackageOverride{
 		p.AddPackage(pkg) // get the rest of the types
 	},
 
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured": func(p *Parser, pkg *loader.Package) {
+		p.Schemata[TypeIdent{Name: "Unstructured", Package: pkg}] = apiext.JSONSchemaProps{
+			Type: "object",
+		}
+		p.AddPackage(pkg) // get the rest of the types
+	},
+
 	"k8s.io/apimachinery/pkg/util/intstr": func(p *Parser, pkg *loader.Package) {
 		p.Schemata[TypeIdent{Name: "IntOrString", Package: pkg}] = apiext.JSONSchemaProps{
 			AnyOf: []apiext.JSONSchemaProps{

